semgrep/semgrep/project.py

```diff
@@ -12,17 +13,29 @@
 logger = getLogger(__name__)
 
 
-def get_project_url() -> Optional[str]:
+def get_project_url(targets: Optional[Sequence[str]] = None) -> Optional[str]:
     """
     Returns the current git project's default remote URL, or None if not a git project / no remote
+
+    If multiple targets are specified, only the first will be used to evaluate the git remote URL.
+    If no target is specified, will fall back to using current working directory
     """
+    target = None
+    if targets and len(targets) > 0:
+        if targets and len(targets) > 1:
+            logger.debug(
+                f"only examining directory {targets[0]} to determine project url"
+            )
+        target = targets[0]
+
     try:
         return cast(
             str,
             sub_check_output(
                 ["git", "ls-remote", "--get-url"],
                 encoding="utf-8",
                 stderr=subprocess.DEVNULL,
+                cwd=target,
```

Comment:
I don't think this is right? Does this work if target is not a directory?

Reply:
you're right, it throws `NotADirectoryError` if it's a directory